lib/src/executable.dart

```diff
@@ -63,12 +63,8 @@ Future<int> runExecutable(Entrypoint entrypoint, String package,
     log.verbosity = log.Verbosity.WARNING;
   }
 
-  // Snapshots are compiled in Dart 1 mode, so in Dart 2 mode we always run
-  // executables from source.
+  // Uncached packages are run from source.
   if (snapshotPath != null) {
-    // Look for the Dart 2-specific snapshot when running in Dart 2 mode.
-    if (isDart2) snapshotPath += '.dart2';
```

Comment:
I think we need to avoid finding the snapshot if we're not in Dart 2 mode otherwise we're going to attempt to run a Dart 2 snapshot from a Dart 1 vm in some cases.

If we're OK with a hard cutover we could skip this, but it might be nice to have a fallback of running from source.

Reply:
My intent was to do a hard cutover and no longer support pub being run in Dart 1 mode. (I think we'll be making similar hard cut-offs to other packages. As soon as any code starts using `whereType()` or optional `new`, you are no longer in Dart 1 land.)

Reply:
synced offline, we'll recommend avoiding the latest SDK if Dart 1 support is necessary with `pub run`.